BEIMA.Backend.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument { 
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            var result = mongo.InsertDevice(doc);
+            Console.WriteLine(result.ToString());
+            Assert.IsNotNull(result);
+            Assert.That(result, Is.TypeOf(typeof(ObjectId)));
+        }
+
+        [Test]
+        public void DocumentNotInserted_InsertDocumentAndDeleteDocument_DocumentHasBeenDeleted()
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument
+            {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            //Insert device
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Delete device
+            bool deleteResult = false;
+            if(insertResult != null)
+            {
+                deleteResult = mongo.DeleteDevice((ObjectId) insertResult);
+            }
+            Assert.IsTrue(deleteResult);
+        }
+
+        [Test]
+        public void DocumentNotInserted_InsertDocumentAndUpdateDocument_DocumentHasBeenUpdated()
+        {
+            var mongo = MongoConnector.Instance;
+            BsonDocument doc = new BsonDocument
+            {
+                { "deviceTypeId", "a" },
+                { "serialNumber", "a12345"}
+            };
+            //Insert device
+            var insertResult = mongo.InsertDevice(doc);
+            Assert.IsNotNull(insertResult);
+            Assert.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Update device
+            doc.AddRange(new BsonDocument { { "updatedDeviceField", "123" } });
+            var updateResult = mongo.UpdateDevice(doc);
+            Assert.IsTrue(updateResult);
+        }
+
+        [Test]
+        public void ConnectorCreated_GetDeviceGivenInvalidDeviceId_NullReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //This is a valid ObjectId, but this is not in the database
+            var doc = mongo.GetDevice(new ObjectId("61f4882f42dab933544849d3"));
+            Assert.IsNull(doc);
+        }
+
+        [Test]
+        public void ConnectorCreated_DeleteInvalidObject_FalseReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //This is a valid ObjectId, but this is not in the database
+            var result = mongo.DeleteDevice(new ObjectId("61f4882f42dab933544849d3"));
+            Assert.IsFalse(result);
+        }
+
+        [Test]
+        public void ConnectorCreated_InsertNull_NullReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            var result = mongo.InsertDevice(null);
+            Assert.IsNull(result);
```

Comment:
Do we want NULL to be returned if it is inserted or should we throw an exception?

Reply:
I wanted null to be returned so we don't have to wrap everything in our endpoints to be wrapped in try/catch blocks. Instead, the exception is abstracted out and we have something easier to check.

Reply:
I think discussing how we want to do exception handling would be a good thing to bring up in the retro as well.